pandas/core/indexes/period.py

```diff
@@ -54,19 +53,49 @@ def f(self):
     f.__doc__ = fget.__doc__
     return property(f)
 
-
 # --- Period index sketch
 
 
 def _new_PeriodIndex(cls, **d):
     # GH13277 for unpickling
-    if d['data'].dtype == 'int64':
-        values = d.pop('data')
-    return cls._from_ordinals(values=values, **d)
+    values = d.pop('data')
+    if values.dtype == 'int64':
+        return cls._from_ordinals(values=values, **d)
+    else:
+        return cls(values, **d)
 
 
-class PeriodIndex(PeriodArrayMixin, DatelikeOps, DatetimeIndexOpsMixin,
-                  Int64Index):
+class PeriodDelegateMixin(PandasDelegate):
+    """
+    Delegate from PeriodIndex to PeriodArray.
+    """
+    def _delegate_property_get(self, name, *args, **kwargs):
+        result = getattr(self._data, name)
+        box_ops = (
+            set(PeriodArray._datetimelike_ops) - set(PeriodArray._bool_ops)
+        )
+        if name in box_ops:
+            result = Index(result, name=self.name)
+        return result
+
+    def _delegate_property_set(self, name, value, *args, **kwargs):
+        setattr(self._data, name, value)
+
+    def _delegate_method(self, name, *args, **kwargs):
+        result = operator.methodcaller(name, *args, **kwargs)(self._data)
+        return Index(result, name=self.name)
+
+
+@delegate_names(PeriodArray,
+                PeriodArray._datetimelike_ops + ['size', 'asi8', 'shape'],
+                typ='property')
+@delegate_names(PeriodArray,
+                [x for x in PeriodArray._datetimelike_methods
+                 if x not in {"asfreq", "to_timestamp"}],
+                typ="method",
+                overwrite=True)
+class PeriodIndex(DatelikeOps, DatetimeIndexOpsMixin,
+                  Int64Index, PeriodDelegateMixin):
```

Comment:
Are people OK with PeriodIndex also being an accessor? That seemed like the easiest way to dispatch things down to PeriodArray.

Reply:
Is performance affected?  I'm still in the "inheritance is the easiest way to dispatch" camp, but recognize that I've lost this one.

Reply:
Beyond an extra function call (or two), I don't think so, though I will check. These aren't descriptors so we aren't changing getitem or getattribute.

Reply:
we could make a generic mechanism to do this at some point. We are eventually going to be doing this with all Index.

Reply:
Related to that, as I think this is for a follow-up, how do we keep track of the follow-up ideas? Update a list in the top-post (to not have it somewhere in the middle be hidden by github)